server/src/main/java/org/elasticsearch/search/RescoreDocIds.java

```diff
@@ -0,0 +1,51 @@
+/*
+ * Licensed to Elasticsearch under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.elasticsearch.search;
+
+import org.elasticsearch.common.io.stream.StreamInput;
+import org.elasticsearch.common.io.stream.StreamOutput;
+import org.elasticsearch.common.io.stream.Writeable;
+
+import java.io.IOException;
+import java.util.Map;
+import java.util.Set;
+
+public final class RescoreDocIds implements Writeable {
```

Comment:
Would be nice to add java documentation describing why is the purpose of this class?

 Is the goal if some shards fail between querying and fetching phase, a coordinating node can continue fetching phase using other shards, including explanation of rescoring where this class would be necessary?

Reply:
I've added a javadocs in https://github.com/elastic/elasticsearch/pull/56480/commits/aeeee70b15305fa9a7450f25a3efe1a8a3216bba.